README.md

```diff
@@ -80,24 +83,184 @@ After you input your application credentials, you can click the setup button, wh
+
+If your carrier is not supported yet by MobiledgeX it will throw a DmeDnsException and it will connect you to  [wifi.dme](https://developers.mobiledgex.com/getting-started/connecting-client-app#distributed-matching-engine) which will connect you to the closest [regional DME](https://developers.mobiledgex.com/getting-started/connecting-client-app#distributed-matching-engine).
 
 ```csharp
 using MobiledgeX;
+using DistributedMatchEngine;
+using UnityEngine;
+using System.Collections;
+
+[RequireComponent(typeof(MobiledgeX.LocationService))]
+public class YourClassName : MonoBehaviour
+{ 
+    IEnumerator Start()
+    {
+        yield return StartCoroutine(MobiledgeX.LocationService.EnsureLocation()); // Location is needed to connect you to the closet edge
+        GetEdgeConnection();
+    }
+     async void GetEdgeConnection()
+    {
+        MobiledgeXIntegration mxi = new MobiledgeXIntegration();
+        try
+        {
+            await mxi.RegisterAndFindCloudlet();
+        }
+        catch(DmeDnsException)
+        {
+            mxi.UseWifiOnly(true); // if you carrier is not supported yet, WifiOnly will connect you to wifi.dme 
+            await mxi.RegisterAndFindCloudlet();
+        }
+
+        mxi.GetAppPort(LProto.L_PROTO_HTTP); // Get the port of the desired protocol
+        string url = mxi.GetUrl("http"); // Get the url of the desired protocol
+    }
+    
+}
+```
+
+If your device doesn't have MCC-MNC code (no sim card - for ex. Oculus device), Please use UseWifiOnly before RegisterAndFindCloudlet.
+
+```csharp
+use mxi.UseWifiOnly(true); 
+await mxi.RegisterAndFindCloudlet();
+```
+
+
+
+
 
-MobiledgeXIntegration integration = new MobiledgeXIntegration();
-bool registered = await integration.Register(); //calls Register Client
-DistributedMatchEngine.FindCloudletReply findCloudletReply = await integration.FindCloudlet(); //calls Find Cloudlet
+**In UnityEditor** 
 
-MatchingEngine dme = integration.me;
+While developing in Unity Editor (Location is not used),The fallback location by default is San Jose,CA.
 
-Dictionary<int, AppPort> appPortsDict = dme.GetHTTPAppPorts(findCloudletReply);
-int public_port = findCloudletReply.ports[0].public_port; // if you only have one port
-AppPort appPort = appPortsDict[public_port];
-HttpClient http = await dme.GetHTTPClient(findCloudletReply, appPort, public_port, 5000);
-HttpResponseMessage message = await http.GetAsync("/"); //makes a get request
+If you wish to change the fallback Location, use SetFallbackLocation() before you call RegisterAndFindCloudlet().
+
+```csharp
+ mxi.SetFallbackLocation(testLongtiude, testLatitude); 
+ await mxi.RegisterAndFindCloudlet();
 ```
 
+By default in Unity Editor you will be Wifi Mode which will connect you to wifi.dme.
```

Comment:
maybe insead of using Wifi Mode which could be confused for UseWifiOnly. this line can be updated to

you will connect with the Wifi DME, which is specified using the TestCarrierInfoClass in the CarrierInfoIntegration script